go/libraries/doltcore/sqle/read_replica_database.go

```diff
@@ -230,6 +223,58 @@ func pullBranches(ctx *sql.Context, rrd ReadReplicaDatabase, branches []string)
 	return nil
 }
 
+func getReplicationBranches(ctx *sql.Context, rrd ReadReplicaDatabase) ([]string, []ref.DoltRef, error) {
+	srcRefs, err := rrd.srcDB.GetBranches(ctx)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	localRefs, err := rrd.Database.ddb.GetBranches(ctx)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	allBranches := make([]string, 0, len(srcRefs))
+	deletedBranches := make([]ref.DoltRef, 0, len(localRefs))
+	var i, j int
+	for i < len(srcRefs) && j < len(localRefs) {
+		rem := srcRefs[i].GetPath()
+		local := localRefs[j].GetPath()
+		if rem == local {
+			allBranches = append(allBranches, rem)
+			i++
+			j++
+		} else if rem < local {
+			allBranches = append(allBranches, rem)
+			i++
+		} else {
+			deletedBranches = append(deletedBranches, localRefs[j])
+			j++
+		}
+	}
+	for i < len(srcRefs) {
+		allBranches = append(allBranches, srcRefs[i].GetPath())
+		i++
+	}
+	for j < len(localRefs) {
+		deletedBranches = append(deletedBranches, localRefs[i])
```

Comment:
`i` is never being incremented here, this is suspicious

This algorithm could use a quick unit test (deleted branches at the beginning, at the end, in the middle)

Reply:
yeah that aint right, thanks